packages/expo/src/hooks/useLocalCredentials/shared.ts

```diff
@@ -0,0 +1,31 @@
+import type { SignInResource } from '@clerk/types';
+
+type LocalCredentials = {
+  identifier?: string;
+  password: string;
+};
+
+type BiometricType = 'fingerprint' | 'face-recognition';
+
+type LocalCredentialsReturn = {
+  setCredentials: (creds: LocalCredentials) => Promise<void>;
+  hasCredentials: boolean;
+  userOwnsCredentials: boolean | null;
+  clearCredentials: () => Promise<void>;
+  authenticate: () => Promise<SignInResource>;
+  biometryType: BiometricType | null;
+};
+
+const LocalCredentialsInitValues: LocalCredentialsReturn = {
```

Comment:
❓ Should we have something like `isPlatformSupported` in `LocalCredentialsInitValues` in order to be easier to check if the hook can be used, as now it seems you will have to check if `biometryType` is null which is not so clear if you don't take a look at the code

Reply:
For devices that do not support biometric, android and iOS fallback to PIN, and if a pin does not exist then `biometricType` will be null.

with biometricType you get the available authenticator and it is useful mostly for UI purposes. Displaying a Face ID or a touch ID icon for example.

Maybe simply changing the name to `supportedBiometricType` would do ?

We don't wanna indicate that the device can use biometric, but instead if we can use local authentication (biometric or not) in order to store credentials.

Reply:
I was mostly referring for platforms that this will not work at all like web,the biometric support was just an example as I didn't know if this fallbacks to pin